src/main/scala/abalone/variant/Variant.scala

```diff
@@ -102,6 +103,144 @@ abstract class Variant private[variant] (
         )
     }.toMap
 
+  def validMovesOf2(situation: Situation, validMoves1: Map[Pos, List[Move]]): Map[Pos, List[(String, Move)]] =
```

Comment:
some comments to understand the cases could be helpful, it was a little tricky to follow through first time.

Reply:
but given the tests work, its not strictly necessary.